modules/tensor_mechanics/include/materials/cohesive_zone_model/SalehaniIrani3DCTraction.h

```diff
@@ -0,0 +1,40 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "CZMComputeLocalTractionTotal.h"
+
+/**
+ * Implementation of the non-stateful exponential traction separation law
+ * proposed by Salehani, Mohsen Khajeh and Irani, Nilgoon 2018
+ **/
+class SalehaniIrani3DCTraction : public CZMComputeLocalTractionTotal
```

Comment:
Since this traction separation law works for both 2D and 3D, maybe we should remove `3D` from its name. User might think it is 3D only.

Reply:
All traction separation laws can be used in 1D, 2D and 3D, but hey always must eb coded for the 3D case. In this specific case, 3DC is part of the model name, as it is an extension of the Xu-Needleman model to the 3D case

Reply:
what does `3DC` stand for?

Reply:
3DC stands for 3D contact, and is the name the authors gave to the model.  Actually I was wrong about the origin, this model is an extension of the McGarry et al.  So I have no particular preference, which itself is an extension of the model from Bosch et al.

Reply:
If authors gave the name to the model and called it in their paper, I guess 3DC is fine to use. No change is needed.

Reply:
ok, it wills tay as is